UAL-RE/ReM
# Add form module to intake of metadata

Closes #6

Note: At the moment, testing is only possible with public datasets.
- [x] Testing with public dataset
    - [x] Ensure that metadata is saved
    - [x] Ensure that editing is possible with previous submitted data populated on re-take
- [x] Ensure 404 page is returned if invalid `article_id` is provided

Commit history:
- Add additional dependencies
- Add intake_form module
- Add intake page
- Specify form name in post method
- Add temporary database
- intake_post: Issue with using IntakeData model. Use dict
- Add type hinting for hello_world get
- Pass submit_dict to intake.html
- Handle IntakeData model
- Update database when entry is present
- Add tinydb in requirements.txt
- Use tinydb for database management in intake_form
- Add db_file as optional input (for unit testing)
- Add unit test for intake_form module
- Add read_form unit test
- Add unit test for intake_form.update_data
- Add unit tests for intake_form.intake_post
- Add db_file to intake_post method for testing
- Add pydantic VersionModel
- Update docstrings for intake_form [ci skip]
- Define tinydb_file for simplicity
- Simplify URL routing with readme_url_path
- Simplify unit tests for forms with for loop
- main: Enable reload for uvicorn run
- Fix routing for form get and post calls
- Add receive.html
- Fix unit tests for URL routing changes
- Mount /templates path for static content (css, js)
- Add + and - symbols for accordion
- Update receive.html to use static css and js
- Add table in second accordion for result metadata
- Redirect to 404 for incorrect article_id in read_form and intake_post methods
- Fix HTML indentation (convention) [ci skip]
- Italicize ReDATA note [ci skip]
- Adjust heading for preferred citation [ci skip]
-  Adjust DOI section for language [ci skip]
- Add indentation for figshare metadata [ci skip]
- Adjust content of Preferred Citation section [ci skip]
- Adjust DOI section [ci skip]
- Add additional sections in intake.html
- Add form response for preferred citation
- Add top navigation bar
- Make top navbar fixed [ci skip]
- Adjust indentations for intake.html [ci skip]
- Add anchor class for padding
- Paragraph setting for fs.summary [ci skip]
- Remove results dict, not needed [ci skip]
- intake_post: Make all form responses optional with default: empty
- Ensure links open new tab [ci skip]
- Define fs_meta div class
- Make form responses optional
- Simplify: change usrform to form in intake.html [ci skip]
- Fix files and folder sections in intake.html [ci skip]
- Fix styling for ReDATA name [ci skip]
- Fix optional text in heading for each section [ci skip]
- Revised receive.html
- Add content for contributor roles section in intake.html [ci skip]
- Add ReDATA logo
- Pass in fields dict for full name for intake_post()
- Add content of references after Additional Notes section [ci skip]
- Change topnav hover color [ci skip]
- Change main.py to support multiple apps on different ports
